src/main/java/org/prebid/server/bidder/dmx/DmxBidder.java

```diff
@@ -62,39 +62,32 @@ public DmxBidder(String endpointUrl, JacksonMapper mapper) {
 
     @Override
     public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
-        final List<BidderError> errors = new ArrayList<>();
-        final List<Imp> validImps = new ArrayList<>();
-
-        if (request.getUser() == null) {
-            if (request.getApp() == null) {
-                errors.add(BidderError.badInput("No user id or app id found. Could not send request to DMX."));
-                return Result.of(Collections.emptyList(), errors);
-            }
+        if (request.getUser() == null && request.getApp() == null) {
+            return Result.emptyWithError(BidderError
+                    .badInput("No user id or app id found. Could not send request to DMX."));
         }
 
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> validImps = new ArrayList<>();
         final List<Imp> imps = request.getImp();
-        if (CollectionUtils.isEmpty(imps)) {
-            errors.add(BidderError.badInput("No valid impressions in the bid request"));
-            return Result.of(Collections.emptyList(), errors);
-        }
 
         String updatedPublisherId = null;
         String updatedSellerId = null;
-        for (int i = 0; i < imps.size(); i++) {
-            try {
-                final ExtImpDmx extImp = parseImpExt(imps.get(i));
-                if (i == 0) {
-                    final String publisherId = extImp.getPublisherId();
-                    updatedPublisherId = StringUtils.isBlank(publisherId)
-                            ? publisherId
-                            : extImp.getMemberId();
-                    updatedSellerId = extImp.getSellerId();
+        try {
+            final ExtImpDmx extImp = parseImpExt(imps.get(0));
+            final String publisherId = extImp.getPublisherId();
+            updatedPublisherId = StringUtils.isBlank(publisherId)
+                    ? publisherId
+                    : extImp.getMemberId();
+            updatedSellerId = extImp.getSellerId();
+            for (Imp imp : request.getImp()) {
```

Comment:
try need to be inside loop. 
Just create two try catch blocks for extraction of publisher and for imps